### PR TITLE
Extract pose data

### PR DESCRIPTION
**Pull request by matharotheelf**
Added a class which manages the extraction of key pose data needed for the strecth classification process.

The feature data is:
- the displacement betwen the nose and the shoulder mid point (to measure head movement)
- the dispacement between the shoulder and elbows and elbows and hands for arm movement

This data is then converted into a single feature array which can be provided to a machine learning model for classification.

### CODE REVIEW

File: sketch.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ let video;
 let bodyPose;
 let poses = [];
 let connections;
+let timeAccumalator = 0;
```
Comment (edennisCS) [inline sketch.js:13]: I think better that accumalator should be spelt like accumulator

File: sketch.js [hunk 2]
```diff
@@ -34,6 +36,13 @@ function draw() {
   // Draw the webcam video
   image(video, 0, 0, width, height);
 
+  timeAccumalator += deltaTime;
+
+  if(timeAccumalator >= poseSampleInterval && poses.length == 1) {
```
Comment (edennisCS) [inline sketch.js:44]: maybe strict equality is better === though unlikely to be any issues

File: sketch.js [hunk 3]
```diff
@@ -34,6 +36,13 @@ function draw() {
   // Draw the webcam video
```
Comment (edennisCS) [inline sketch.js:39]: Draw function looks good though could work as multiple sub functions to separate logic (as I saw it has three different stages) not critical

**Comment by edennisCS** [comment]
There are some places with missing semi colons I think it is fine for now though as they are not in critical areas. Maybe can auto format it later or something

**Comment by matharotheelf** [comment]
> There are some places with missing semi colons I think it is fine for now though as they are not in critical areas. Maybe can auto format it later or something

I added eslint and fixed the missing semi colons!

**Review by edennisCS — COMMENTED** [review]
Forgot to push these as review, I also made these suggestions :P

**Pull request merged**